ocaml/typing/uniqueness_analysis.ml

```diff
@@ -1100,6 +1100,18 @@ and pattern_match_single pat paths : Ienv.Extension.t * UF.t =
       |> conjuncts_pattern_match
     in
     ext, UF.par uf_read uf_args
+  | Tpat_unboxed_tuple args ->
+    (* XXX modes *)
+    let uf_read = Paths.mark_implicit_borrow_memory_address Read occ paths in
```

Comment:
This is subtle. `memory_address` here refers to the box memory of a tuple. Unboxed tuples don't have such box so needs to be handled differently.

This file is not exercised in production currently, so it's fine to be loose. I will push a commit of what I think is correct, and let's call it a day.

Reply:
I pushed what I think is correct (with testing showing its difference to regular tuple). Resolving.